src/sql/workbench/parts/charts/browser/actions.ts

```diff
@@ -143,70 +142,50 @@ export class SaveImageAction extends Action {
 	public static ICON = 'saveAsImage';
 
 	constructor(
-		@IWindowsService private windowsService: IWindowsService,
-		@IWindowService private windowService: IWindowService,
-		@INotificationService private notificationService: INotificationService,
-		@IEditorService private editorService: IEditorService,
-		@IWorkspaceContextService private workspaceContextService: IWorkspaceContextService
+		@IWindowsService private readonly windowsService: IWindowsService,
+		@INotificationService private readonly notificationService: INotificationService,
+		@IFileService private readonly fileService: IFileService,
+		@IFileDialogService private readonly fileDialogService: IFileDialogService
 	) {
 		super(SaveImageAction.ID, SaveImageAction.LABEL, SaveImageAction.ICON);
 	}
 
-	public run(context: IChartActionContext): Promise<boolean> {
+	public async run(context: IChartActionContext): Promise<boolean> {
 		if (context.insight instanceof Graph) {
-			return this.promptForFilepath().then(filePath => {
-				let data = (<Graph>context.insight).getCanvasData();
-				if (!data) {
-					this.showError(localize('chartNotFound', "Could not find chart to save"));
-					return false;
-				}
-				if (filePath) {
-					let buffer = this.decodeBase64Image(data);
-					writeFile(filePath, buffer).then(undefined, (err) => {
-						if (err) {
-							this.showError(err.message);
-						} else {
-							let fileUri = URI.file(filePath);
-							this.windowsService.openExternal(fileUri.toString());
-							this.notificationService.notify({
-								severity: Severity.Error,
-								message: localize('chartSaved', 'Saved Chart to path: {0}', filePath)
-							});
-						}
-					});
-				}
-				return true;
-			});
+			const filePath = await this.fileDialogService.pickFileToSave({});
+			const data = (<Graph>context.insight).getCanvasData();
+			if (!data) {
+				this.notificationService.error(localize('chartNotFound', "Could not find chart to save"));
+				return false;
+			}
+			if (filePath) {
+				let buffer = this.decodeBase64Image(data);
+				this.fileService.writeFile(filePath, buffer).then(undefined, (err) => {
+					if (err) {
```

Comment:
Is there a reason you changed the pickFileToSave to be awaited but then left this one as a .then?

Reply:
I didn't rewrite this section, but replace the call for writeFile. I actually rewrote the pick file part. I can change this.